### PR TITLE
gophercloud authentication: set gophercloud.EndpointOpts, update golangci-lint

### PR DESCRIPTION
**Pull request by chrischdi**
* Reads and sets the region name in gophercloud.EndpointOpts
* updates golangci-lint to v1.30.0

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -117,18 +117,19 @@ func main() {
 
 // run does the initialization of the operational exporter and also the metrics scraping
 func run() error {
-	var opts gophercloud.AuthOptions
+	var ao *gophercloud.AuthOptions
+	var eo *gophercloud.EndpointOpts
```
Comment (tobiasgiese) [inline main.go:121]: maybe use some more informative variable names? It's hard to read imo

Comment (tobiasgiese) [reply main.go:121]:
```suggestion
	var authOpts *gophercloud.AuthOptions
	var endpointOpts *gophercloud.EndpointOpts
```

**Review by sbueringer — COMMENTED** [review]
Thx for the comment @tobiasgiese otherwise /lgtm :D

File: main.go [hunk 2]
```diff
@@ -117,18 +117,19 @@ func main() {
...
 	if *cloudConfFile != "" {
-		opts, err = authOptsFromCloudConf(*cloudConfFile)
+		ao, eo, err = authOptsFromCloudConf(*cloudConfFile)
 		if err != nil {
 			return logError("unable to read OpenStack credentials from cloud.conf: %v", err)
 		}
 		klog.Infof("OpenStack credentials read from cloud.conf file at %s", *cloudConfFile)
 	} else {
-		opts, err = openstack.AuthOptionsFromEnv()
+		ao, eo, err = openstack.AuthOptionsFromEnv()
```
Comment (tobiasgiese) [inline main.go:132]:
```
vet: ./main.go:132:17: cannot assign 2 values to 3 variables
```

**Review by tobiasgiese — APPROVED** [review]

**Pull request merged**